src/devices/device_models/common/duration_constraints.jl

```diff
@@ -145,19 +303,17 @@ function device_duration_param(ps_m::CanonicalModel,
         for i in (t-duration_data[ix].down + 1):t
             if t <= duration_data[ix].down
                 if in(i, time_steps)
-                    JuMP.add_to_expression!(lhs_off, (1-ps_m.variables[var_names[1]][name, i]))
+                    JuMP.add_to_expression!(lhs_off, (1-varon[name, i]))
                 end
             else
-                JuMP.add_to_expression!(lhs_off, ps_m.variables[var_names[3]][name, i])
+                JuMP.add_to_expression!(lhs_off, varstop[name, i])
             end
         end
         if t <= duration_data[ix].down
-            lhs_off += initial_duration_on[ix].value
-            ps_m.constraints[name_down][name, t] = JuMP.@constraint(ps_m.JuMPmodel,
-             ps_m.variables[var_names[2]][name, t]*duration_data[ix].down - lhs_off <= 0.0)
+            lhs_off += initial_duration_on[ix].value #typo? off not on?
```

Comment:
Yes, that should be `initial_duration_off[ix].value`. Could you make the change.
Thanks

Reply:
The change has been made!